src/ApiCore/ValidationException.php

```diff
@@ -32,10 +32,11 @@
 namespace Google\ApiCore;
 
 use Exception;
+use RuntimeException;
 
 /**
  * ValidationException represents a local error (i.e. not during an RPC call).
  */
-class ValidationException extends Exception
+class ValidationException extends RuntimeException
```

Comment:
I agree with your reasoning that `RuntimeException` is preferable to `ApiException`.

Reply:
There are [some cases](https://github.com/googleapis/gax-php/blob/0de75cc58255a13e00ea87e85d2c0e6ea2b234e2/src/ApiCore/RetrySettings.php#L288) we use the `ValidationException` as a substitute for an `InvalidArgumentException`.  In the [SPL exception hierarchy](http://php.net/manual/en/spl.exceptions.php#spl.exceptions.tree) `InvalidArgumentException` and `RuntimeException` belong to different paths (`LogicException` vs `RuntimeException`). Does it seem like it might be worthwhile to keep that separation in our library as well? Perhaps it could mean adding another exception which we use for invalid type cases?

Reply:
This is an interesting point. Is it the case that we are substituting for an `InvalidArgumentException` here? Or are we substituting for an `UnexpectedValueException`? Even so, are there other places in our code where we are using `ValidationException` in a way that we should really be using a `LogicException`?

Reply:
Good point, that example does seem to align more with an `UnexpectedValueException`. [This](https://github.com/googleapis/gax-php/blob/master/src/ApiCore/GapicClientTrait.php#L326-L329) seems to be a better example of what could be a `LogicException` or `InvalidArgumentException`.

Also, I could see an argument for something like [this](https://github.com/googleapis/gax-php/blob/master/src/ApiCore/OperationResponse.php#L207) to be a `BadMethodCallException` (which would fall under `LogicException`). WDYT?

Reply:
I'm not sure about that first example either. I feel like it falls under the category of "you have provided a bad `$options` argument", which is sort of what `ValidationException` is meant to cover - bad inputs from the user. I do think there is a strong case that the exception in your second example should be something else, but I don't think `BadMethodCallException` - from the decription: "Exception thrown if a callback refers to an undefined method or if some arguments are missing.", but this is more like you have called this method at the wrong time.

Any though, my main feeling is that reasonable people can disagree on this issue :-)

Reply:
I have reverted the change - I would rather err on the side on not churning. Maybe we can discuss some time next week to see if there are any really strong reasons to make this change (my reasons for making it are fairly weak ones) but I don't want to hold up this PR until then.